db/data_migration/20150407095340_convert_policies_to_html_publications.rb

```diff
@@ -0,0 +1,147 @@
+module PolicyMigrationHelpers
+  def self.build_html_attachment_body(policy, url_maker, alternative_format_provider)
+    callout = %|
+$CTA
+This is a copy of a document that stated a policy of the 2010 to 2015 Conservative and Liberal Democrat coalition government.
+The previous URL of this page was [#{url_maker.public_document_url(policy)}](#{url_maker.public_document_url(policy)})
+Current policies can be found at the GOV.UK [policies list](#{url_maker.policies_url}).
+$CTA
+
+|
+
+    combined_body = callout + policy.body
+
+    supporting_pages = policy.supporting_pages.published
+    supporting_pages.each_with_index do |supporting_page, index|
+      combined_body += extract_supporting_page(supporting_page, index, alternative_format_provider)
+    end
+
+    return combined_body
+  end
+
+  def self.copy_associations!(policy, policy_paper)
+    [
+      :lead_organisations,
+      :supporting_organisations,
+      :topics,
+      :topical_events,
+      :nation_inapplicabilities,
+      :role_appointments,
+      :fact_check_requests,
+      :world_locations,
+      :related_documents,
+      :specialist_sectors,
+    ].each do |association|
+      policy_paper.send("#{association}=".to_sym, policy.send(association))
+    end
+  end
+
+  def self.extract_supporting_page(supporting_page, index, alternative_format_provider)
+    appendix_title = "Appendix #{index+1}: #{supporting_page.title}"
+
+    puts "-- Adding supporting page ##{supporting_page.id}: as #{appendix_title}"
+
+    sp_body = %|
+
+###{appendix_title}
+
+$CTA
+This was a supporting detail page of the main policy document.
+$CTA
+
+#{supporting_page.body.gsub(/^(\s*[#]{2,})/, '\1#')}
+|
+
+    sp_body.gsub(/(?:^!@([0-9]+)|\[InlineAttachment:([0-9]+)\])/) do
+      if attachment = supporting_page.attachments[$1.to_i - 1]
```

Comment:
Here's an illustration of the result (the publication that is generated from Policy 225886): 

![screen shot 2015-04-16 at 14 25 16](https://cloud.githubusercontent.com/assets/3687/7181898/c1723518-e445-11e4-8e0e-8d8ebe851499.png)